### PR TITLE
gRPC server side - error reports (Fix Bugsnag)

### PR DESCRIPTION
**Pull request by darrenclark**
Crashes weren't being picked up by [bugsnag-elixir](https://github.com/jarednorman/bugsnag-elixir)

- Updated `GRPC.Adapter.Cowboy.Handler` to use `:proc_lib.spawn_link/3` so that error reports get logged

- When sending back a `"Deadline expired"` response to the client, kicked off a dummy process so an error gets reported

### CODE REVIEW

File: test/grpc/integration/server_test.exs [hunk 1]
```diff
@@ -115,22 +119,45 @@ defmodule GRPC.Integration.ServerTest do
...
+
+  test "return errors for unknown errors in library code" do
+    attach_error_handler()
+
+    run_server([HelloErrorServer], fn port ->
+      {:ok, channel} = GRPC.Stub.connect("localhost:#{port}")
+      req = Helloworld.HelloRequest.new(name: "bad protobuf")
+
+      assert {:error,
+              %GRPC.RPCError{message: "Internal Server Error", status: GRPC.Status.unknown()}} ==
```
Comment (nicocharlery) [inline test/grpc/integration/server_test.exs:144]: 👍

File: lib/grpc/adapter/cowboy/handler.ex [hunk 2]
```diff
@@ -36,7 +36,7 @@ defmodule GRPC.Adapter.Cowboy.Handler do
         compressor: compressor
       }
 
-      pid = spawn_link(__MODULE__, :call_rpc, [server, path, stream])
+      pid = :proc_lib.spawn_link(__MODULE__, :call_rpc, [server, path, stream])
```
Comment (nicocharlery) [inline lib/grpc/adapter/cowboy/handler.ex]: For my knowledge, why is this change necessary?

**Review by nicocharlery — APPROVED** [review]

File: lib/grpc/adapter/cowboy/handler.ex (hunk 2 above)
Comment (darrenclark) [reply lib/grpc/adapter/cowboy/handler.ex]: it didn't quite work as i had hoped, so doing another approach

Comment (darrenclark) [reply lib/grpc/adapter/cowboy/handler.ex]: `:proc_lib` makes it a special process http://erlang.org/doc/design_principles/spec_proc.html

part of that includes generating error reports when a process crashes, i.e. https://github.com/erlang/otp/blob/d9516215a82c9297180da0196f0e1e43cfd26535/lib/stdlib/src/proc_lib.erl#L521-L532

And, Bugsnag can listen for those crashes (https://github.com/jarednorman/bugsnag-elixir/blob/525231a514e4627f00112064734942b4b32e4388/lib/bugsnag.ex#L20-L22) and report them.

`:proc_lib.spawn_link` is essentially a wrapper around `spawn_link` that does some extra things to make it a special process.

Generally better to use higher-level modules like `Task`, `GenServer`, `:gen_statem`, etc. instead of `:proc_lib` though.

(I didn't reach for `Task` here because I wanted to avoid doing as much surgery to the module as possible.)

Comment (nicocharlery) [reply lib/grpc/adapter/cowboy/handler.ex]: Thanks much for the explanation 👍

**Review by Corvillus — APPROVED** [review]
